app/src/main/java/com/example/playlistmaker/di/DataModule.kt

```diff
@@ -36,28 +37,47 @@ val dataModule = module {
 
     single<HistoryRepository> {
         HistoryRepositoryImpl(
-            sharedPrefsRepository = get(),
-            gsonRepository = get()
+            sharedPreferences = get(),
+            gson = get()
         )
     }
 
-    single<GsonRepository> {
-        GsonRepositoryImpl()
+    single<TrackRepository> {
+        TrackRepositoryImpl(RetrofitNetworkClient(context = get(), retrofit = get()))
     }
 
-    single<TrackRepository> {
-        TrackRepositoryImpl(RetrofitNetworkClient(context = get(), retrofitRepository = get()))
+    single<PlayerRepository> {
+        PlayerRepositoryImpl(get())
     }
 
-    single<RetrofitRepository>{
-        RetrofitRepositoryImpl(get(), interceptorClientRepository = get())
+    single<SharedPreferences> {
+        androidApplication().getSharedPreferences(
+            PLAY_LIST_SHARED_PREFERENCES,
+            Context.MODE_PRIVATE
+        )
     }
 
-    single<InterceptorClientRepository>{
-        InterceptorClientRepositoryImpl()
+    single<Gson> {
```

Comment:
🍏 Gson не хранит никаких специальных настроек и стейта, поэтому можно каждый раз при необходимости создавать новый объект и не хранить его все время жизни модуля.
```suggestion
    factory<Gson> {
```

Reply:
А медиаплеера это касается? factory<MediaPlayer>